lib/ruby_terraform/command_line/option.rb

```diff
@@ -0,0 +1,60 @@
+require 'json'
+
+module RubyTerraform
+  module CommandLine
+    class Option
+      def initialize(opt_key, value, switch_override: nil)
+        assign_option_value(value)
+        @switch = switch_override || switch_from_key(opt_key)
+      end
+
+      def add_to_subcommand(sub)
+        if value.respond_to?(:keys)
+          add_hash_to_subcommand(sub, value)
+        elsif value.respond_to?(:each)
+          add_array_to_subcommand(sub, value)
+        else
+          add_option_to_subcommand(sub)
+        end || sub
+      end
+
+      private
+
+      attr_reader :switch, :value
+
+      def assign_option_value(value)
+        @value = value
+      end
+
+      def switch_from_key(opt_key)
+        "-#{opt_key.to_s.sub('_', '-')}"
+      end
+
+      def add_hash_to_subcommand(sub, hash)
+        hash.each do |hash_key, hash_value|
+          sub = sub.with_option(switch, "'#{hash_key}=#{as_string(hash_value)}'", separator: ' ')
+        end
+        sub
+      end
+
+      def add_array_to_subcommand(sub, array)
+        array.each do |element|
+          sub = sub.with_option(switch, element)
+        end
+        sub
+      end
+
+      def add_option_to_subcommand(sub)
+        if value
+          sub.with_option(switch, value)
+        else
+          sub
+        end
+      end
```

Comment:
Wondering if this too belongs in `lino`. Is there ever a case where you'd expect `nil` to be added as a swtich value? I think if there was a command that expected it, it would be reasonable to expect it to be passed as a string `'nil'`. I think I've convinced myself that it should go into `lino` in the `#with_option` method.